site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,327 @@
+| :------------- | :---------- |
+| *ACCEPTED*     | Indicates that Hono has successfully received and accepted the request for processing. |
+| *REJECTED*     | Indicates that Hono has received the request but was not able to process it. The *error* field contains information regarding the reason why. Clients should not try to re-send the request using the same message properties in this case. |
+
+# Payload Format
+
+Most of the operations of the Credentials API allow or require the inclusion of payload data as part of the
+request or response messages of the operation. Such payload is carried in the body of the corresponding AMQP 
+messages as part of a single *AMQP Value* section.
+
+The payload consists of a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The table below provides an overview of the standard members defined for the JSON object:
+
+| Name             | Mandatory | Type       | Default Value | Description |
+| :--------------- | :-------: | :--------- | :------------ | :---------- |
+| *device-id*      | *yes*     | *string*   |               | The ID of the device to which the credentials belong. |
+| *type*           | *yes*     | *string*   |               | The credential type name. The value may be arbitrarily chosen by clients but SHOULD reflect the particular type of authentication mechanism the credentials are to be used with. Possible values include (but are not limited to) `psk`, `RawPublicKey`, `hashed-password` etc. |
+| *auth-id*        | *yes*     | *string*   |               | The identity that the device should be authenticated as. |
+| *enabled*        | *no*      | *boolean*  | *true*        | If set to *false* the credentials are not supposed to be used to authenticate devices any longer. **NB** It is up to the discretion of the protocol adapter to make use of this information. |
+| *not-before*     | *no*      | *string*   | *null*        | The point in time from which on the credentials may be used to authenticate devices. If not *null*, the value MUST be an [ISO 8601 compliant *combined date and time representation*](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations). **NB** It is up to the discretion of the protocol adapter to make use of this information. |
+| *not-after*      | *no*      | *string*   | *null*        | The point in time until which the credentials may be used to authenticate devices. If not *null*, the value MUST be an [ISO 8601 compliant *combined date and time representation*](https://en.wikipedia.org/wiki/ISO_8601#Combined_date_and_time_representations). **NB** It is up to the discretion of the protocol adapter to make use of this information. |
+
+For each set of credentials the combination of `auth-id` and `type` MUST be unique within a tenant.
+
+The JSON object MAY contain additional members of arbitrary other names which MUST be of a scalar type only.
```

Comment:
@sophokles73 : regarding your concerns about bootstrapping: isn't that a problem outside of Hono? If we automatically disable a password after e.g. a certain day is reached, the problem how a second, now valid password was installed or transfered to the device is not Hono's business, is it?
I just thought it is necessary to have such time period support for validity, including the described support for consecutive, adjacent validity periods, but that we do not care about how devices switch their passwords.
In certain scenarios it may be appropriate to have passwords with unlimited validity (test scenarios e.g.), but this is possible with the defined API then as well.
That is my current understanding at least, so I would still vote for the _enabled_, _not-before_, _not-after_ properties. WDYT?